src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,369 @@
+  public Pane contentPane;
+  public BarChart<String, Integer> ratingDistributionChart;
+  public Label playerDivisionNameLabel;
+  public Label playerScoreLabel;
+  public Label scoreLabel;
+  public Label seasonLabel;
+  public ComboBox<SubdivisionBean> majorDivisionPicker;
+  public Arc scoreArc;
+  public TabPane subDivisionTabPane;
+  public ImageView playerDivisionImageView;
+  public CategoryAxis xAxis;
+  public NumberAxis yAxis;
+  public Label placementLabel;
+  public Label seasonDateLabel;
+  private LeagueSeasonBean season;
+  private LeagueEntryBean entryToSelect;
+  private InvalidationListener playerRatingListener;
+
+  @VisibleForTesting
+  protected AutoCompletionBinding<String> usernamesAutoCompletion;
+
+  @Override
+  public void initialize() {
+    scoreLabel.setText(i18n.get("leaderboard.score").toUpperCase());
+    searchTextField.setPromptText(i18n.get("leaderboard.searchPrompt").toUpperCase());
+
+    JavaFxUtil.bindManagedToVisible(contentPane);
+    JavaFxUtil.bindManagedToVisible(connectionProgressPane);
+    connectionProgressPane.visibleProperty().bind(contentPane.visibleProperty().not());
+
+    majorDivisionPicker.setConverter(divisionStringConverter());
+    yAxis.setTickLabelFormatter(integerStringConverter());
+
+    searchTextField.textProperty().addListener((observable, oldValue, newValue) ->
+        processSearchInput(newValue));
+
+    subDivisionTabPane.widthProperty().addListener((observable, oldValue, newValue) ->
+        setTabWidth((double) newValue, subDivisionTabPane.getTabs().size()));
+  }
+
+  private void setTabWidth(double tabPaneWidth, int tabNumber) {
+    // The tabs always appear 40px wider, maybe because of some obscure padding rule.
+    // We also need to subtract the tab area padding to prevent horizontal scrolling.
+    double tabWidth = Math.floor((tabPaneWidth - 20.0) / tabNumber) - 40.0;
+    JavaFxUtil.runLater(() -> {
+      subDivisionTabPane.setTabMinWidth(tabWidth);
+      subDivisionTabPane.setTabMaxWidth(tabWidth);
+    });
+  }
+
+  private void processSearchInput(String searchText) {
+    playerService.getPlayerByName(searchText).thenAccept(playerBeanOptional -> playerBeanOptional.ifPresent(player ->
+      leaderboardService.getLeagueEntryForPlayer(player, season.getId()).thenAccept(leagueEntry -> {
+        if (leagueEntry == null) {
+          return;
+        }
+        selectLeagueEntry(leagueEntry);
+      })));
+  }
+
+  public void setSeason(LeagueSeasonBean season) {
+    this.season = season;
+
+    seasonLabel.setText(i18n.getOrDefault(season.getNameKey(), String.format("leaderboard.season.%s", season.getNameKey())).toUpperCase());
+    String startDate = season.getStartDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+    String endDate = season.getEndDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+    seasonDateLabel.setText(i18n.get("leaderboard.seasonDate", startDate, endDate));
```

Comment:
Where can the user select a date format?

Reply:
use TimeService.asDate and it is set in the SettingsController

Reply:
There are two issues I see with this: The setting is explicitely in the chat settings, so I would expect it to only affect the chat. Secondly, I want the dateformat to be medium (Jan 5, 2021) instead of short (01/05/2021), so I am not convinced that using the timeservice functionality here is a good idea

Reply:
That setting is no longer chat only and just needs to be moved in the settings ui. As far as the dateformat length then I would just modify the asDate function to take in the formatStyle as a parameter as well. That way the date formats can be consistent as the user set them according to the locale they want across all times